plugins/rhacm-backend/src/helpers/parser.test.ts

```diff
@@ -0,0 +1,169 @@
+        value: 'https://console.cluster.1cloud',
+      },
+      {
+        name: 'region.open-cluster-management.io',
+        value: 'eu',
+      },
+
+      {
+        name: 'oauthredirecturis.openshift.io',
+        value: 'https://oauth.cluster1.cloud',
+      },
+    ],
+    conditions: [
+      {
+        message: 'Managed cluster is available',
+        status: 'True',
+        type: 'ManagedClusterConditionAvailable',
+      },
+    ],
+  },
+};
+
+const PARSED_CLUSTER: ClusterDetails = {
+  allocatableResources: {
+    cpuCores: 94.5,
+    memorySize: '656645580Ki',
+    numberOfPods: 750,
+  },
+  availableResources: {
+    cpuCores: 96,
+    memorySize: '660098508Ki',
+    numberOfPods: 750,
+  },
+  consoleUrl: 'https://console.cluster.1cloud',
+  kubernetesVersion: 'v1.22.0-rc.0+a44d0f0',
+  name: 'cluster1',
+  oauthUrl: 'https://oauth.cluster1.cloud',
+  openshiftId: '24ee1f0a-c8ac-4833-83ac-dccec690825c',
+  openshiftVersion: '4.9.5',
+  platform: 'BareMetal',
+  region: 'eu',
+  status: {
+    available: true,
+    reason: 'Managed cluster is available',
+  },
+};
+
+const CLAIMS = {
+  status: {
+    clusterClaims: [
+      {
+        name: 'claim1',
+        value: 'claim1_value',
+      },
+      {
+        name: 'claim2',
+        value: 'claim2_value',
+      },
+      {
+        name: 'claim3',
+        value: 'claim3_value',
+      },
+    ],
+  },
+};
+
+describe('getClaim', () => {
+  it('should extract a cluster claim value from a cluster object', () => {
+    const cluster = CLAIMS;
+
+    expect(getClaim(cluster, 'claim2')).toStrictEqual('claim2_value');
+  });
+
+  it("should return undefined when cluster claim doesn't exist", () => {
+    const cluster: any = CLAIMS;
+    cluster.status.clusterClaims[1] = {};
+
+    const result = getClaim(cluster, 'claim2');
+
+    expect(result).toBeUndefined();
+  });
+});
+
+describe('parseManagedCluster', () => {
+  it('should parse a managed cluster to cluster details', () => {
+    const cluster = UNPARSED_CLUSTER;
+
+    const result = parseManagedCluster(cluster);
+
+    const expected = PARSED_CLUSTER;
+
+    expect(result).toStrictEqual(expected);
+  });
+
+  it('should parse a managed cluster without labels to cluster details', () => {
+    const cluster: any = UNPARSED_CLUSTER;
+    cluster.metadata.labels = {};
+
+    const result = parseManagedCluster(cluster);
+
+    const expected = PARSED_CLUSTER;
+
+    expect(result).toStrictEqual(expected);
+  });
+
+  it('should parse an unavaliable managed cluster to cluster details', () => {
+    const cluster = UNPARSED_CLUSTER;
+    cluster.status.conditions = [
+      {
+        message: 'Managed cluster is unavailable',
+        status: 'False',
+        type: 'ManagedClusterConditionAvailable',
+      },
+    ];
+
+    const result = parseManagedCluster(cluster);
+
+    const expected = PARSED_CLUSTER;
+    expected.status.available = false;
+    expected.status.reason = 'Managed cluster is unavailable';
```

Comment:
same as above. Modifying a referenced object introduces nondeterminism. You can either do partial compares against constant/literal data or make a copy of the object.

Reply:
Also.. Always thing about the test case purpose. Do you really want to match the whole parsed cluster every time? Or is the test case interested in specific property only? Why make it harder for yourself by comparing the whole object, if you want to assert on single property only? This can also lead to false positive test failures - imagine the object structure changes and out of a sudden all the test cases start failing  for every property, even for those which haven't change. And you don't want that. You want each test failure to pinpoint a specific problem in the tested unit of code.

Reply:
Thanks for the explanation! Resolved in the latest commit.